main.go

```
@@ -99,6 +99,11 @@ func (c *Collector) collectResource(ch chan<- prometheus.Metric, resource string
 
 // Collect - collect results from Azure Montior API and create Prometheus metrics.
 func (c *Collector) Collect(ch chan<- prometheus.Metric) {
+	if err := ac.refreshAccessToken(); err != nil {
+		log.Println(err)
```

You should fail the query here by putting an invalid metric into the channel